client/packages/common/src/ui/forms/JsonForms/useJsonForms.tsx

```diff
@@ -265,33 +154,49 @@ export const useJsonForms = (
     </Box>
   );
 
+  useEffect(() => {
+    if (!databaseResponse) return;
+
+    const { data, documentRegistry } = databaseResponse;
+    if (!data) {
+      setError('No document data');
+    } else {
+      setData(data);
+      setDocumentId(databaseResponse.id);
+    }
+
+    if (!documentRegistry) {
+      setError('No document registry entry');
+    } else if (!documentRegistry.jsonSchema) {
+      setError('No Json Schema');
+    } else if (!documentRegistry.uiSchema) {
+      setError('No UI Schema');
+    } else {
+      setDocumentRegistry(documentRegistry);
+    }
+  }, [databaseResponse]);
+
+  // user createDoc if there is one
+  useEffect(() => {
+    if (createDoc) {
+      setData(createDoc.data);
+      setDocumentRegistry(createDoc.documentRegistry);
+      setIsDirty(true);
+    }
+  }, [createDoc]);
+
   return {
     JsonForm: (
-      <Box
-        id="document-display"
-        display="flex"
-        flexDirection="column"
-        justifyContent="flex-start"
-        alignItems="center"
-        width="100%"
-        gap={2}
-        paddingX={10}
+      <JsonForm
+        data={data}
+        documentRegistry={documentRegistry}
+        isError={isError}
+        isLoading={isLoading}
+        setError={setError}
+        updateData={updateData}
       >
-        <ScrollFix />
-        {isLoading || !data ? (
-          <BasicSpinner />
-        ) : (
-          <FormComponent
-            data={data}
-            jsonSchema={documentRegistry.jsonSchema}
-            uiSchema={documentRegistry.uiSchema}
-            setData={updateData}
-            setError={setError}
-            renderers={renderers}
-          />
-        )}
-        {showButtonPanel && !isLoading && <ButtonPanel />}
-      </Box>
+        {showButtonPanel && <ButtonPanel />}
```

Comment:
the change with the `ButtonPanel` was to display when loading to avoid a bit of jank, and have them disabled instead.